src/actions/vault/VaultSPTActions.sol

```diff
@@ -293,7 +293,7 @@ contract VaultSPTActions is Vault20Actions {
 
         // approve Sense Finance Periphery to transfer underliers on behalf of proxy
         if (swapParams.approve != 0) {
-            IERC20(swapParams.assetIn).approve(address(periphery), swapParams.approve);
+            IERC20(swapParams.assetIn).safeApprove(address(periphery), swapParams.approve);
```

Comment:
@SolcAndMe Here we set allowance for `swapParams.approve` but for the `swapUnderlyingForPTs` we use underlierAmount, if `swapParams.approve` > `underlierAmount` then we need to reset the allowance next time( set to 0 before we can set it to a new value). Shouldn`t we set the allowance to `underlierAmout`?

Reply:
wdym?

Reply:
For example when we call `buyCollateralAndModifyDebt()` we pass the `underlierAmount` as a parameter but we also have `swapParams` which contains the `approve` value. And in `_buyPToken()` we set allowance(for the periphery contract) for `swapParams.approve` but pass `underlierAmount` when calling `periphery.swapUnderlyingForPTs()` which in turn will transfer the underlier. So if somehow what we pass in the swap struct is different than underlierAmount we could revert because of insufficient allowance or reversely revert next time because we will set a new allowance and the current allowance is non-zero (in case of usdt).

Reply:
so, since user may set allowance to some arbitrary value by manually configuring `swapParams.approve` and needs to update in a later tx, before `safeApprove` to the new value we have to check whether `allowance > 0` in which case we need to first reset `safeApprove` to 0. this is an additional check that needs to be done within the `if(swapParams.approve != 0)` check

Reply:
lgtm now

Reply:
> For example when we call `buyCollateralAndModifyDebt()` we pass the `underlierAmount` as a parameter but we also have `swapParams` which contains the `approve` value. And in `_buyPToken()` we set allowance(for the periphery contract) for `swapParams.approve` but pass `underlierAmount` when calling `periphery.swapUnderlyingForPTs()` which in turn will transfer the underlier. So if somehow what we pass in the swap struct is different than underlierAmount we could revert because of insufficient allowance or reversely revert next time because we will set a new allowance and the current allowance is non-zero (in case of usdt).

that's the exact behaviour went want

Reply:
that's what's there :eyes: